pyomo/repn/plugins/nl_writer.py

```diff
@@ -1110,13 +1134,13 @@ def write(self, model):
                     else:
                         val = lb if abs(lb) < abs(ub) else ub
                     eliminated_vars[_i] = AMPLRepn(val, {}, None)
-                    _vmap[_i] = expr_info.compile_repn(visitor)[0]
+                    nl_map[_i] = expr_info.compile_repn(visitor)[0]
                     logger.warning(
                         "presolve identified an underdetermined independent "
                         "linear subsystem that was removed from the model.  "
                         f"Setting '{var_map[_i]}' == {val}"
                     )
-            _vmap[_id] = nl.rstrip() % tuple(_vmap[_i] for _i in args)
```

Comment:
Why do you no longer need the `rstrip`?

Reply:
Because the `\n` was removed from `template.var`: this greatly simplified management of newlines when substituting variables with expressions / constants.

Reply:
Fabulous.